### PR TITLE
Adds 'sequence' for Map[A, ErrorOr[B]]

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: src/main/scala/lenthall/validation/ErrorOr.scala [hunk 1]
```diff
@@ -19,4 +21,10 @@ object ErrorOr {
       }
     }
   }
+
+  implicit class MapErrorOrRhs[A, B](val m: Map[A, ErrorOr[B]]) extends AnyVal {
+    private def elementToErrorOr(kvp: (A, ErrorOr[B])): ErrorOr[(A, B)] = kvp._2 map { v => (kvp._1, v) }
+
+    def sequence: ErrorOr[Map[A, B]] = m.toList.traverse[ErrorOr, (A, B)] { x => elementToErrorOr(x) } map { _.toMap }
```
Comment (mcovarr) [inline src/main/scala/lenthall/validation/ErrorOr.scala]: ToL could you implement `traverse` and have `sequence` be a special case that applies `identity` to the elements?

**Review by mcovarr — APPROVED** [review]

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/lenthall/pull-request/43/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/lenthall)

**Comment by cjllanwarne** [comment]
@mcovarr is this what you intended?

**Comment by mcovarr** [comment]
Bravo

**Comment by danbills** [comment]
You should get `sequence` and `traverse` for free (w/ the right import sauce) w/ cats

**Comment by cjllanwarne** [comment]
@danbills I wasn't able to get that to work, even with `cats.implicits._`, although that by no means rules it out as possible...

**Comment by danbills** [comment]
@cjllanwarne I wasn't either, but I _was_ able to get it to work in Cats console... Investigating

cats:
```
scala> import cats.syntax.traverse._
import cats.syntax.traverse._

scala> import cats.instances.map._
import cats.instances.map._

scala> import cats.data.Validated._
import cats.data.Validated._

scala> validNel(1)
res0: cats.data.ValidatedNel[Nothing,Int] = Valid(1)

scala> Map("h" -> res0).sequence
res1: cats.data.ValidatedNel[Nothing,scala.collection.immutable.Map[String,Int]] = Valid(Map(h -> 1))
```

lenthall:
```
scala> import cats.syntax.traverse._
import cats.syntax.traverse._

scala> import cats.instances.map._
import cats.instances.map._

scala> import cats.data.Validated._
import cats.data.Validated._

scala> validNel(1)
res0: cats.data.ValidatedNel[Nothing,Int] = Valid(1)

scala> Map("h" -> res0).sequence
<console>:22: error: value sequence is not a member of scala.collection.immutable.Map[String,cats.data.ValidatedNel[Nothing,Int]]
       Map("h" -> res0).sequence
```

**Review by danbills — APPROVED** [review]

**Pull request merged**